code/datums/hud/critter.dm

```diff
@@ -287,29 +297,33 @@
 					src.master.force_laydown_standup()
 				src.update_resting()
 
-/// updates health hud element
+/// Updates health hud element
 /datum/hud/critter/proc/update_health()
-	if (!isdead(src.master))
-		if (!src.health) //Runtime fix: Cannot modify null.icon_state
-			return
-		var/h_ratio = (src.master.health / src.master.max_health) * 100
-		switch(h_ratio)
-			if(90 to INFINITY)
-				src.health.icon_state = "health0" // green with green marker
-			if(75 to 90)
-				src.health.icon_state = "health1" // green
-			if(60 to 75)
-				src.health.icon_state = "health2" // yellow
-			if(45 to 60)
-				src.health.icon_state = "health3" // orange
-			if(20 to 45)
-				src.health.icon_state = "health4" // dark orange
-			if(10 to 20)
-				src.health.icon_state = "health5" // red
-			else
-				src.health.icon_state = "health6" // crit
-	else
-		src.health.icon_state = "health7"         // dead
+	var/datum/hud_element/health_elem = src.get_hudzone("health_zone").get_element("health")
+	var/atom/movable/screen/hud/screen_obj = health_elem.screen_obj
+	if (isnull(screen_obj))
+		return
+	if (isdead(src.master))
+		screen_obj.icon_state = "health7"		// dead
+		return
+
+	var/h_ratio = (src.master.health / src.master.max_health) * 100
+	switch(h_ratio)
+		if(90 to INFINITY)
+			screen_obj.icon_state = "health0"	// green with green marker
+		if(75 to 90)
+			screen_obj.icon_state = "health1"	// green
+		if(60 to 75)
+			screen_obj.icon_state = "health2"	// yellow
+		if(45 to 60)
+			screen_obj.icon_state = "health3"	// orange
+		if(20 to 45)
+			screen_obj.icon_state = "health4"	// dark orange
+		if(10 to 20)
+			screen_obj.icon_state = "health5"	// red
+		else
+			screen_obj.icon_state = "health6"	// crit
```

Comment:
This is parity with existing, but noting for future discussion that it's weird that you show as "crit" for the last 10% of your health, right? I don't think you're actually in crit at that point.

Reply:
For critters you pretty much are, since you die at 0% instantly. That was how I always read it.